### PR TITLE
github-events: generate BQ schemas for cloudevent-recorder

### PR DESCRIPTION
**Pull request by imjasonh**
This adds structs for known types, with fields we know we care about, starting with `pull_request` and `workflow_run`.

These types then have a BQ schema generated for them, which can be passed to `cloudevent-recorder`:

```
module "github-events" {
  source = "./modules/github-events"

  regions = module.networking.regional-networks
  ingress = module.cloudevent-broker.ingress
  ...
}

module "cloudevent-recorder" {
  source = "./modules/cloudevent-recorder"

  broker = module.cloudevent-broker.broker
  types  = module.github-events.recorder-schemas // <-- this
  ...
}
```

Events of other types get published as-is, but aren't recorded.

### CODE REVIEW

**Comment by imjasonh** [comment]
Screenshot 2024-03-19 at 4 04 33 PM

File: go.mod [hunk 1]
```diff
@@ -2,7 +2,10 @@ module github.com/chainguard-dev/terraform-infra-common
 
 go 1.21.6
 
+replace cloud.google.com/go/bigquery => github.com/imjasonh/google-cloud-go/bigquery v0.0.0-20240311234559-1dc0a176b346
```
Comment (mattmoor) [inline go.mod:5]: What's this?

File: modules/github-events/cmd/schemagen/main.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2022 Chainguard, Inc.
```
Comment (k4leung4) [inline modules/github-events/cmd/schemagen/main.go:2]:
```suggestion
Copyright 2024 Chainguard, Inc.
```

File: test.tf
Comment (mattmoor) [inline test.tf:1]: did you mean to check this in?

File: go.mod (hunk 1 above)
Comment (k4leung4) [inline go.mod:5]: is the plan to upstream your changes or to depend on your fork moving forward
please add a comment

File: .github/workflows/go-test.yaml [hunk 3]
```diff
@@ -47,3 +47,8 @@ jobs:
     - run: |
         # Exclude running unit tests against third_party repos.
         go test -race ./...
+
+    - name: Ensure github-events schemas are up to date
+      run: |
+        go run ./modules/github-events/cmd/schemagen
+        git diff --exit-code
```
Comment (mattmoor) [inline .github/workflows/go-test.yaml:54]: we had a `nodiff` action

File: test.tf
Comment (imjasonh) [reply test.tf:1]: Nope!

**Comment by imjasonh** [comment]
Screenshot 2024-03-19 at 4 36 33 PM

🍿

**Comment by imjasonh** [comment]
Screenshot 2024-03-19 at 4 37 22 PM

🎉 

(mostly; not sure what's up with the empty fields, we'll work on that)

**Review by found-it — APPROVED** [review]
The questions I had were addressed in other comments. I'm dependent on some of the workflow run fields but I'm okay with merging this in so we can iterate on the types.

**Pull request merged**